### PR TITLE
Protocol v6

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

**Review by mlohvynenko — APPROVED** [review]

File: api/cloudprotocol/desiredstatus.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+	BlockIv      []byte `json:"blockIv"`
+	BlockKey     []byte `json:"blockKey"`
+	AsymAlg      string `json:"asymAlg"`
+	ReceiverInfo *struct {
+		Serial string `json:"serial"`
+		Issuer []byte `json:"issuer"`
+	} `json:"receiverInfo"`
+}
+
+// DownloadInfo struct contains how to download item.
```
Comment (mykola-kobets-epam) [inline api/cloudprotocol/desiredstatus.go:114]: it looks like it specifies how to download mulitple items

File: api/cloudprotocol/unitstatus.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+// NodeInfo node information.
+type NodeInfo struct {
+	NodeID     string          `json:"nodeId"`
+	NodeType   string          `json:"nodeType"`
+	Name       string          `json:"name"`
+	Status     string          `json:"status"`
+	CPUs       []CPUInfo       `json:"cpus"`
+	OSTypes    string          `json:"osTypes"`
```
Comment (mykola-kobets-epam) [inline api/cloudprotocol/unitstatus.go:94]: should be OSType I guess

File: api/cloudprotocol/unitstatus.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+	StateChecksum string     `json:"stateChecksum,omitempty"`
+	RunState      string     `json:"runState"`
+	NodeID        string     `json:"nodeId"`
+	ErrorInfo     *ErrorInfo `json:"errorInfo,omitempty"`
+}
+
+// LayerStatus layer status.
+type LayerStatus struct {
+	LayerID   string     `json:"layerId"`
+	Digest    string     `json:"digest"`
```
Comment (mykola-kobets-epam) [inline api/cloudprotocol/unitstatus.go:122]: Shouldnt it be byte array?

File: api/cloudprotocol/unitstatus.go [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+	Version       string     `json:"version"`
+	Status        string     `json:"status"`
+	ErrorInfo     *ErrorInfo `json:"errorInfo,omitempty"`
+}
+
+// UnitStatus unit status structure.
+type UnitStatus struct {
+	MessageType  string             `json:"messageType"`
+	IsDeltaInfo  bool               `json:"isDeltaInfo"`
+	UnitConfig   []UnitConfigStatus `json:"unitConfig"`
```
Comment (mykola-kobets-epam) [inline api/cloudprotocol/unitstatus.go:141]: UnitConfig seems to be a single value

**Review by mykola-kobets-epam — APPROVED** [review]

File: api/cloudprotocol/unitstatus.go (hunk 3 above)
Comment (al1img) [reply api/cloudprotocol/unitstatus.go:122]: No, It contains also digest type: for example: `sha256:5b0bcabd1ed22e9fb1310cf6c2dec7cdef19f0ad69efa1f392e94a4333501270`

File: api/cloudprotocol/desiredstatus.go (hunk 1 above)
Comment (al1img) [reply api/cloudprotocol/desiredstatus.go:114]: Unfortunately or fortunately, this information for one item.

File: api/cloudprotocol/unitstatus.go (hunk 2 above)
Comment (al1img) [reply api/cloudprotocol/unitstatus.go:94]: fixed

File: api/cloudprotocol/unitstatus.go (hunk 4 above)
Comment (al1img) [reply api/cloudprotocol/unitstatus.go:141]: No. We may return multiple items for unit config status:

`[{version: 1, "installed"},{ version: 2, "error"}]`

To indicate that current version 1, installing version 2 fails.

**Review by mykola-kobets-epam — APPROVED** [review]

**Pull request merged**